### PR TITLE
created-profile-model

### PR DESCRIPTION
**Pull request by GurkirtSingh**
### What this PR does (required):
- A Profile model is created to store user's details, location and availability.
- Link to ticket https://hatchways.notion.site/Profile-Model-df539b4588b14eb0b21fb4c878b6ebe6

### CODE REVIEW

File: server/models/Profile.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+const mongoose = require("mongoose");
+
+const profileSchema = new mongoose.Schema({
+  user_id: {
```
Comment (ugur-akin) [inline server/models/Profile.js]: We generally prefer `camelCase` for field names in JS: for example `user_id` would become `userId` and `firstname` would be `firstName`. Let's fix all the properties accordingly 👍🏼

File: server/models/Profile.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
```
Comment (ugur-akin) [inline server/models/Profile.js]: There shuould be a `ref` SchemaType here as well ([relevant mongoose docs](https://mongoosejs.com/docs/populate.html#populate)) 👍🏼

File: server/models/Profile.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    },
+    province: {
+      type: String,
+      required: true,
+    },
+    country: {
+      type: String,
+      required: true,
+    },
+    index: true,
```
Comment (ugur-akin) [inline server/models/Profile.js:46]: I like the idea🙌🏼 Although I'm personally not 💯 if `index` works in nested paths, would be a nice exercise to find out 🤔

File: server/models/Profile.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        required: true,
+        index: true,
+      },
+      end_at: {
+        type: Date,
+        required: true,
+      },
+    },
+  ],
+  hourlyWage: {
```
Comment (ugur-akin) [inline server/models/Profile.js]: This is subtle but I don't think `wage` is the best choice of word for this. Maybe `hourlyRate` or `hourlyFee` instead? What do you think?

**Review by ugur-akin — COMMENTED** [review]
Good start 🚀 The model looks good overall, there are just a few minor issues to address.

I noticed we didn't actualy compile and/or export a model from the schema defined here. I suggest taking a look at the [docs](https://mongoosejs.com/docs/models.html#compiling) and add the code for those as well 👍🏼 

As a bonus, it would have been nice to some validation as well. Just to give some ideas, we could set some boundries on the birth date, we could check `end > start` for availability blocks or we could use regex validation for url/phone number.

File: server/models/Profile.js (hunk 1 above)
Comment (GurkirtSingh) [reply server/models/Profile.js]: Alright I'll follow camelCase pattern for fields from now on.

File: server/models/Profile.js (hunk 2 above)
Comment (GurkirtSingh) [reply server/models/Profile.js]: Agree with you. I'll fix it

File: server/models/Profile.js (hunk 4 above)
Comment (GurkirtSingh) [reply server/models/Profile.js]: Yeah that was confusing name 😃. Will fix it.

**Comment by GurkirtSingh** [comment]
Thank for the review @ugur-akin. Yes, it was mistake to not export the model I will do it in next commit. Also I'll add validation for the fields. 👨🏽‍🔧

File: server/models/Profile.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+const availabilityEndAtValidator = function (value) {
+  return value > this.startAt;
+};
+
+profileSchema
+  .path("phoneNumber")
+  .validate(phoneNumberValidator, "`{VALUE}` is not valid phone number");
+profileSchema
+  .path("photoUrl")
+  .validate(photoUrlValidator, "`{VALUE}` is not valid photo url");
```
Comment (moffatethan) [inline server/models/Profile.js:86]: 💯

**Review by moffatethan — APPROVED** [review]
LGTM

**Comment by GurkirtSingh** [comment]
Perfect, then 👨🏽‍💻 I'll merge it.

**Pull request merged**